mailbox/api/src/main/scala/org/apache/james/mailbox/scala/ScalaQuotaEvent.scala

```diff
@@ -0,0 +1,108 @@
+package org.apache.james.mailbox.scala
+
+import java.time.Instant
+import java.util.Optional
+
+import julienrf.json.derived
+import org.apache.james.core.quota.{QuotaCount, QuotaSize, QuotaValue}
+import org.apache.james.core.{Domain, User}
+import org.apache.james.mailbox.MailboxListener.{QuotaEvent => JavaQuotaEvent, QuotaUsageUpdatedEvent => JavaQuotaUsageUpdatedEvent}
+import org.apache.james.mailbox.model.{QuotaRoot, Quota => JavaQuota}
+import play.api.libs.json._
+
+import scala.collection.JavaConverters._
+
+
+sealed trait ScalaQuotaEvent {
+  def getQuotaRoot: QuotaRoot
+  def toJava: JavaQuotaEvent
+}
+
+case class Quota[T <: QuotaValue[T]](used: T, limit: T, limits: Map[JavaQuota.Scope, T]) {
+  def toJava: JavaQuota[T] =
+    JavaQuota.builder[T].used(used).computedLimit(limit).limitsByScope(limits.asJava).build()
+}
+
+case class QuotaUsageUpdatedEvent(quotaRoot: QuotaRoot, countQuota: Quota[QuotaCount],
+                                  sizeQuota: Quota[QuotaSize], time: Instant) extends ScalaQuotaEvent {
+  override def getQuotaRoot: QuotaRoot = quotaRoot
+  override def toJava: JavaQuotaEvent =
+    new JavaQuotaUsageUpdatedEvent(null, getQuotaRoot, countQuota.toJava, sizeQuota.toJava, time)
+}
+
+case class QuotaNoop(user: User, quotaRoot: QuotaRoot) extends ScalaQuotaEvent {
+  override def getQuotaRoot: QuotaRoot = quotaRoot
+  override def toJava: JavaQuotaEvent = ???
+}
+
+object JsonSerialize {
+  implicit val userWriters: Writes[User] = (user: User) => JsString(user.asString)
```

Comment:
You spoke about something allowing dealing with read & writes at the same time.

Did you investigate.

Reply:
That's really easy, you have to implement Format.
you still have to implement both methods (read and write) so there's no gain in this case, it's useful mostly when you use default Writes/Reads because you write a single line instead of two.